src/tests/system/tests/test_passkey.py

```diff
@@ -0,0 +1,92 @@
+from __future__ import annotations
+
+import pytest
+from sssd_test_framework.roles.client import Client
+from sssd_test_framework.roles.generic import GenericProvider
+from sssd_test_framework.roles.ipa import IPA
+from sssd_test_framework.topology import KnownTopology
+
+
+@pytest.mark.tier(0)
+@pytest.mark.topology(KnownTopology.Client)
+def test_passkey__register__sssctl(client: Client, moduledatadir: str, testdatadir: str):
+    """
+    :title: Register a key with sssctl
+    :setup:
+        1. Setup IDM client with FIDO and umockdev setup
+    :steps:
+        1. Use sssctl to register a FIDO2 key.
+        2. Check umockdev-run that the output contains the user key mapping data.
+    :expectedresults:
+        1. New key is registered
+        2. Output contains key mapping data.
+    :customerscenario: False
+    """
+    mapping = client.sssctl.passkey_register(username="user1",
+                                             domain="ldap.test",
+                                             pin=123456,
+                                             device=f"{moduledatadir}/umockdev.device",
+                                             ioctl=f"{moduledatadir}/umockdev.ioctl",
+                                             script=f"{testdatadir}/umockdev.script",
+                                             )
+
+    with open(f"{testdatadir}/passkey-mapping") as f:
+        assert mapping == f.read().strip()
+
+
+@pytest.mark.tier(0)
+@pytest.mark.topology(KnownTopology.IPA)
+def test_passkey__register__ipa(ipa: IPA, moduledatadir: str, testdatadir: str):
+    """
+    :title: Register a key with IPA sssctl command
+    :setup:
+        1. Setup IDM client with FIDO and umockdev setup
+    :steps:
+        1. Use ipa command to register a FIDO2.
+        2. Check umockdev-run that the output contains the user key mapping data.
```

Comment:
I think you should rephrase this line as you aren't using umockdev directly and it can confuse